shell/browser/api/atom_api_web_contents.cc

```diff
@@ -124,15 +124,34 @@
 
 namespace gin {
 
+std::string GetPrinterDescription(int status) {
+  switch (status) {
+    case 0:
+      return "idle";
+    case 1:
+    case 2:
+      return "error";
+    case 3:
+      return "offline";
+    case 4:
+      return "in use";
+    case 5:
+      return "paused";
```

Comment:
Where did you find the meaning of these codes?

Reply:
they're not documented anywhere unfortunately - i manually tested all of them in order to determine the statuses they corresponded to. This was done via triggering local printers, checking statuses via system preferences ui:

<img width="636" alt="Screen Shot 2019-11-11 at 6 01 37 PM" src="https://user-images.githubusercontent.com/2036040/68609535-69da7800-04ad-11ea-90be-8a8a5174d9ed.png">

and then mapping those statuses to the numbers that was being returned in the object.

Reply:
- Do these have the same meaning on macOS, Windows and Linux?
- Is this an exhaustive list?
- Do we want to become the documentation for this?

If no to any of the above, it seems like it might be better for us (though obviously a worse API) to just pass through the number directly and pass the buck to the app developer to interpret the status code.

Reply:
primarily - i changed this in service of wanting to move away from APIs that essentially just dump implementation details with no filtering or parsing. What would you think about trying to confirm this with Chromium developers to ensure that my current statuses are correct?

Reply:
Yeah, it'd be great to get some clarity on this. I'm down with providing a better API but I want to be careful about how we do it. e.g. i have no idea if any of these status codes might have multiple or different meanings in other situations...

Reply:
i talked to some folks who gave me some more context and informed me that windows and unix use different meanings, so i've instead opted to retain the numeric value and instead link to locations users can go to parse meanings from the integers :)